up-transport-vsomeip/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use tokio::sync::RwLock;
+use up_rust::{UCode, UListener, UStatus, UUri, UUID};
+
+mod determine_message_type;
+mod listener_registry;
+mod message_conversions;
+mod rpc_correlation;
+mod transport_inner;
+use transport_inner::UPTransportVsomeipInner;
+mod vsomeip_config;
+mod vsomeip_offered_requested;
+
+pub mod transport;
+
+// TODO: use function from up-rust when merged
+pub(crate) fn any_uuri() -> UUri {
+    UUri {
+        authority_name: "*".to_string(),
+        ue_id: 0x0000_FFFF,     // any instance, any service
+        ue_version_major: 0xFF, // any
+        resource_id: 0xFFFF,    // any
+        ..Default::default()
+    }
+}
+
+// TODO: upstream into up-rust
+pub(crate) fn any_uuri_fixed_authority_id(authority_name: &AuthorityName, ue_id: UeId) -> UUri {
+    UUri {
+        authority_name: authority_name.to_string(),
+        ue_id: ue_id as u32,
+        ue_version_major: 0xFF, // any
+        resource_id: 0xFFFF,    // any
+        ..Default::default()
+    }
+}
+
+// TODO: upstream this into up-rust
+pub(crate) fn split_u32_to_u16(value: u32) -> (u16, u16) {
+    let most_significant_bits = (value >> 16) as u16;
+    let least_significant_bits = (value & 0xFFFF) as u16;
+    (most_significant_bits, least_significant_bits)
+}
+
+// TODO: upstream this into up-rust
+pub(crate) fn split_u32_to_u8(value: u32) -> (u8, u8, u8, u8) {
+    let byte1 = (value >> 24) as u8;
+    let byte2 = (value >> 16 & 0xFF) as u8;
+    let byte3 = (value >> 8 & 0xFF) as u8;
+    let byte4 = (value & 0xFF) as u8;
+    (byte1, byte2, byte3, byte4)
+}
+
+// TODO: upstream this into up-rust
+pub(crate) fn create_request_id(client_id: ClientId, session_id: SessionId) -> RequestId {
+    ((client_id as u32) << 16) | (session_id as u32)
+}
+
+type ApplicationName = String;
+type AuthorityName = String;
+type UeId = u16;
+type ClientId = u16;
+type ReqId = UUID;
+type SessionId = u16;
+type RequestId = u32;
+type EventId = u16;
+type ServiceId = u16;
+type InstanceId = u16;
+type MethodId = u16;
+
+pub struct UPTransportVsomeip {
+    inner_transport: UPTransportVsomeipInner,
+    authority_name: AuthorityName,
+    remote_authority_name: AuthorityName,
+    ue_id: UeId,
+    config_path: Option<PathBuf>,
+    // if this is not None, indicates that we are in a dedicated point-to-point mode
+    point_to_point_listener: RwLock<Option<Arc<dyn UListener>>>,
+}
+
+impl UPTransportVsomeip {
+    pub fn new_with_config(
+        authority_name: &AuthorityName,
+        remote_authority_name: &AuthorityName,
```

Comment:
Added some documentation to these functions.

From my reading of the [SOME/IP uP-L1 spec](https://github.com/eclipse-uprotocol/up-spec/blob/main/up-l1/someip.adoc#21-uuris):

![image](https://github.com/user-attachments/assets/1c230dc1-4301-47bd-8793-941e5a64abba)

We should set these statically since vsomeip doesn't expose them.

I think it may be worthwhile to have an option to set from the vsomeip config, so I'll go ahead a leave this one open.

Reply:
Created issue #9 to track this work for now. Will mark as resolved and track over there.